donkeycar/parts/datastore_v2.py

```diff
@@ -280,7 +280,7 @@ def write_record(self, record):
         self.current_index += 1
         # Update metadata to keep track of the last index
         self._update_catalog_metadata(update=True)
-        self._got_updated = True
+        self._updated_session = True
```

Comment:
This only needs to be happen when a new session is created right ?
Why are we doing this on every `write_record` ?

Reply:
I meant something like:

```python
if not self._updated_session:
    # Ensure session metadata is persisted when the `Tub` is closed.
    self._updated_session = True
```

Reply:
If we set this to True when the tub is created, we would append the `_session_id` to the metadata, even if no record had been added. In fact we would only need to do this when `write_record()` is called once, but programatically, I don't think we'll gain any performance benefit of doing a check before that. But let me add this for clarity, I think that's what you were asking for.